File: solutions/julia/raindrops/raindrops.jl
using OrderedCollections

drops = OrderedDict(3 => "Pling", 5 => "Plang", 7 => "Plong")

function raindrops(number)
    result = ""
    for dropfactor in keys(drops)
        if number % dropfactor == 0
            result = result * get(drops, dropfactor, "")
        end
    end
    if isempty(result)
        result = string(number)
    end
    return result
end

File: solutions/r/raindrops/raindrops.R
key <- data.frame(div = c(3, 5, 7), name = c("Pling", "Plang", "Plong"))

raindrops <- function(number) {
  sounds <- ifelse(number %% key$div == 0, key$name, "")
  sound <- paste0(sounds, collapse = "")
  if (sound == "") as.character(number) else sound
}
